src/ai/backend/manager/api/vfolder.py

```diff
@@ -294,9 +294,17 @@ def _wrapper(
     ) -> Callable[Concatenate[web.Request, P], Awaitable[web.Response]]:
         @functools.wraps(handler)
         async def _wrapped(request: web.Request, *args: P.args, **kwargs: P.kwargs) -> web.Response:
-            folder_name = request.match_info["name"]
+            folder_name_or_id: str | uuid.UUID
+            piece = request.match_info["name"]
+            try:
+                folder_name_or_id = uuid.UUID(piece)
+            except ValueError:
+                folder_name_or_id = piece
             return await handler(
-                request, await resolve_vfolder_rows(request, perm, folder_name), *args, **kwargs
+                request,
+                await resolve_vfolder_rows(request, perm, folder_name_or_id),
+                *args,
+                **kwargs,
```

Comment:
How about parsing folder id value to UUID in `resolve_vfolder_rows()` rather than in `with_vfolder_rows_resolved()`?
There are some handler functions that call `resolve_vfolder_rows()`

Reply:
Either way is possible, but I thought raw input interpretation should be done in the handler layer while type-based branching should be done inside the logic layer.